huaweicloud/services/eip/resource_huaweicloud_vpc_bandwidth.go

```diff
@@ -173,33 +186,37 @@ func resourceVpcBandWidthV2Create(ctx context.Context, d *schema.ResourceData, m
 	if d.Get("charging_mode").(string) == "prePaid" {
 		// we can not create a bandwidth with pre-paid directly due to the API does not support
 		// call the change-to-period API as a workaround
-		bssClient, err := cfg.BssV2Client(region)
+		err := changeBandwidthToPeriod(ctx, networkingClient, bssClient, d)
 		if err != nil {
-			return diag.Errorf("error creating BSS v2 client: %s", err)
+			return err
 		}
+	}
 
-		changeOpts := bandwidths.ChangeToPeriodOpts{
-			BandwidthIDs: []string{b.ID},
-			ExtendParam: structs.ChargeInfo{
-				ChargeMode:  d.Get("charging_mode").(string),
-				PeriodType:  d.Get("period_unit").(string),
-				PeriodNum:   d.Get("period").(int),
-				IsAutoRenew: d.Get("auto_renew").(string),
-				IsAutoPay:   "true",
-			},
-		}
-		orderID, err := bandwidths.ChangeToPeriod(networkingClient, changeOpts).Extract()
-		if err != nil {
-			return diag.Errorf("error changing bandwidth (%s) to pre-paid billing mode: %s",
-				b.ID, err)
-		}
+	return resourceVpcBandWidthV2Read(ctx, d, meta)
+}
 
-		if err := common.WaitOrderComplete(ctx, bssClient, orderID, d.Timeout(schema.TimeoutCreate)); err != nil {
-			return diag.FromErr(err)
-		}
+func changeBandwidthToPeriod(ctx context.Context, networkingClient *golangsdk.ServiceClient,
+	bssClient *golangsdk.ServiceClient, d *schema.ResourceData) diag.Diagnostics {
```

Comment:
suggest reordering the parameters
```suggestion
func changeBandwidthToPeriod(ctx context.Context, d *schema.ResourceData,
	networkingClient, bssClient *golangsdk.ServiceClient) diag.Diagnostics {
```